bulk/bulk.go

```diff
@@ -37,11 +51,48 @@ func (r *Resource) CreateJob(options Options) (*Job, error) {
 	return job, nil
 }
 
-// AllJobs will retrieve all of the bulk 2.0 jobs.
-func (r *Resource) AllJobs(parameters Parameters) (*Jobs, error) {
-	jobs, err := newJobs(r.session, parameters)
+// JobsInfo will retrieve all of the bulk 2.0 jobs.
+func (r *Resource) JobsInfo(parameters Parameters) ([]Response, error) {
+	jobs, err := jobsInfo(r.session, parameters)
 	if err != nil {
 		return nil, err
 	}
 	return jobs, nil
 }
+
+// QueryJobsResults ...
+func (r *Resource) QueryJobsResults(jobs []*Job, writers []io.Writer, parameters Parameters, waitMaxDuration time.Duration, maxRecords int) (map[string]error, error) {
+
+	if len(jobs) != len(writers) {
+		return map[string]error{}, fmt.Errorf("len(jobs) %d != len(writes) %d", len(jobs), len(writers))
+	}
+
+	errsMap := make(map[string]error)
+	jobsMap := make(map[string]*Job)
+	writersMap := make(map[string]io.Writer)
+	for i, j := range jobs {
+		jobsMap[j.info.ID] = jobs[i]
+		writersMap[j.info.ID] = writers[i]
+	}
+
+	return errsMap, Retry(Backoff{Initial: time.Second, Multiplier: 2, Max: 5 * time.Minute}, func() (bool, error) {
```

Comment:
It would be more desirable to allow for a user to specify the `Retry` 
The compromise would be to have a default backoff provided so that user have the option of using that.

Reply:
OK